src/qrcode.js

```diff
@@ -0,0 +1,330 @@
+        fn = drawModuleRounded;
+    }
+
+    context.beginPath();
+    for (row = 0; row < moduleCount; row += 1) {
+        for (col = 0; col < moduleCount; col += 1) {
+            const l = settings.left + col * moduleSize;
+            const t = settings.top + row * moduleSize;
+            const w = moduleSize;
+
+            fn(qr, context, settings, l, t, w, row, col);
+        }
+    }
+
+    context.fillStyle = settings.fill;
+    context.fill();
+}
+
+function createQRCode(text, level, version, quiet) {
+    const qr = {};
+
+    const vqr = new VendorQrCode(version, level);
+    vqr.addData(text);
+    vqr.make();
+    quiet = quiet || 0;
+
+    const qrModuleCount = vqr.getModuleCount();
+    const quietModuleCount = vqr.getModuleCount() + 2 * quiet;
+
+    function isDark(row, col) {
+        row -= quiet;
+        col -= quiet;
+
+        if (row < 0 || row >= qrModuleCount || col < 0 || col >= qrModuleCount) {
+            return false;
+        }
+        return vqr.isDark(row, col);
+    }
+
+    function addBlank(l, t, r, b) {
+        const prevIsDark = qr.isDark;
+        const moduleSize = 1 / quietModuleCount;
+
+        qr.isDark = function (row, col) {
+            const ml = col * moduleSize;
+            const mt = row * moduleSize;
+            const mr = ml + moduleSize;
+            const mb = mt + moduleSize;
+
+            return prevIsDark(row, col) && (l > mr || ml > r || t > mb || mt > b);
+        };
+    }
+
+    qr.text = text;
+    qr.level = level;
+    qr.version = version;
+    qr.moduleCount = quietModuleCount;
+    qr.isDark = isDark;
+    qr.addBlank = addBlank;
+
+    return qr;
+}
+
+function createMinQRCode(text, level, minVersion, maxVersion, quiet) {
+    minVersion = Math.max(1, minVersion || 1);
+    maxVersion = Math.min(40, maxVersion || 40);
+    for (let version = minVersion; version <= maxVersion; version += 1) {
+        try {
+            return createQRCode(text, level, version, quiet);
+        } catch (err) { /* empty */ }
+    }
+    return undefined;
+}
+
+function drawOnCanvas(element, options, imageElement = null) {
+    const settings = { ...options, image: imageElement };
+    console.log('[CHECKING ELEMENT IN DRAW CANVAS] : ', element, options, imageElement);
+
+    element.width = settings.size;
+    element.height = settings.size;
+
+    const qr = createMinQRCode(settings.text, settings.ecLevel, settings.minVersion, settings.maxVersion, settings.quiet);
+
+    if (!qr) {
+        return null;
+    }
+
+    const context = element.getContext('2d');
+
+    drawBackground(qr, context, settings);
+    drawModules(qr, context, settings);
+
+    return element;
+}
+
+function loadImage(element, options) {
+    const { image } = options;
+    const imageElement = new Image();
+
+    imageElement.addEventListener('load', () => drawOnCanvas(element, options, imageElement));
+    imageElement.addEventListener('error', () => {
+        throw new Error('Please provide a valid image');
```

Comment:
console log the error from image in order to let the integrator know what is the source of the issue